### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by untergeek**
This is the first commit, for review.

### CODE REVIEW

File: DEVELOPER.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# logstash-filter-example
+Example filter plugin. This should help bootstrap your effort to write your own filter plugin!
```
Comment (suyograo) [inline DEVELOPER.md:2]: Change?

File: lib/logstash/filters/de_dot.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    @logger.debug? and @logger.debug("replacement field reference", :newref => newref)
+    event[newref] = event[fieldref]
+    @logger.debug? and @logger.debug("event with both new and old field references", :event => event.to_hash.to_s)
+    event.remove(fieldref)
+    @logger.debug? and @logger.debug("postprocess", :event => event.to_hash.to_s)
+  end
+
+  public
+  def filter(event)
+    @separator = '][' if @nested
```
Comment (suyograo) [inline lib/logstash/filters/de_dot.rb:51]: Make this `] [`a const?

File: DEVELOPER.md (hunk 1 above)
Comment (untergeek) [inline DEVELOPER.md:2]: Oh, my!  I thought it was blank.  Yes, I'll change that.

File: lib/logstash/filters/de_dot.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+
+  public
+  def register
+    # Add instance variables, if any
+  end # def register
+
+  private
+  def rename_field(event, fieldref)
+    @logger.debug? and @logger.debug("preprocess", :event => event.to_hash.to_s)
```
Comment (suyograo) [inline lib/logstash/filters/de_dot.rb:39]: use `&&` instead of `and`

File: lib/logstash/filters/de_dot.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    @logger.debug? and @logger.debug("source field reference", :fieldref => fieldref)
```
Comment (suyograo) [inline lib/logstash/filters/de_dot.rb:40]: use `&&` instead of `and`

File: lib/logstash/filters/de_dot.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    newref = fieldref.gsub('.', @separator)
+    @logger.debug? and @logger.debug("replacement field reference", :newref => newref)
```
Comment (suyograo) [inline lib/logstash/filters/de_dot.rb:42]: same as above

File: lib/logstash/filters/de_dot.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    event[newref] = event[fieldref]
+    @logger.debug? and @logger.debug("event with both new and old field references", :event => event.to_hash.to_s)
```
Comment (suyograo) [inline lib/logstash/filters/de_dot.rb:44]: same as above

File: lib/logstash/filters/de_dot.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    event.remove(fieldref)
+    @logger.debug? and @logger.debug("postprocess", :event => event.to_hash.to_s)
+  end
+
+  public
+  def filter(event)
+    @separator = '][' if @nested
+    @logger.debug? and @logger.debug("Replace dots with separator", :separator => @separator)
```
Comment (suyograo) [inline lib/logstash/filters/de_dot.rb:52]: use `&&` instead of `and`

**Comment by untergeek** [comment]
Something weird happened when I pushed, so I did a new PR.  Closing this one.

**Pull request closed without merging**